pydantic/functional_serializers.py

```diff
@@ -227,3 +227,34 @@ def dec(f: Callable[..., Any]) -> _decorators.PydanticDescriptorProxy[Any]:
         return dec
     else:
         return dec(__f)  # type: ignore
+
+
+AnyType = TypeVar('AnyType')
+
+
+if TYPE_CHECKING:
+    SerializeAsAny = Annotated[AnyType, ...]  # SerializeAsAny[list[str]] will be treated by type checkers as list[str]
+    """Force serialization to ignore whatever is defined in the schema and instead ask the object
+    itself how it should be serialized.
+    In particular, this means that model subclasses will include fields present in the subclass but not in the
+    original schema.
+    """
+else:
+
+    @_internal_dataclass.slots_dataclass
+    class SerializeAsAny:  # noqa: D101
```

Comment:
@lig moving this wasn't necessarily related to adding `Examples`, I'm happy to split up the PR if you'd like